kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/TaxReporting1099ServiceImpl.java

```diff
@@ -0,0 +1,1806 @@
+
+					extractVendorNonPdpPayments(vendor, taxYear, taxYearStartDate, taxYearEndDate, eh, replaceData);
+
+					// if we have added payments then increment payees extracted
+					if (eh.getPaymentsExtracted().intValue() > saveExtractedPaymentCount) {
+						eh.incrementPayeesExtracted();
+					}
+				}
+			}
+
+			// save the ExtractHistory
+			businessObjectService.save(eh);
+
+			retval = true;
+
+			LOG.info("end extractPayees(): " + new Date());
+		} 
+		catch (Exception ex) {
+			throw new RuntimeException("1099 Payee Master Extract Error: ", ex);
+
+		}
+		return retval;
+	}
+
+	private void extractVendorPdpPayments(VendorDetail vendor, Integer taxYear, Timestamp taxYearStartDate, Timestamp taxYearEndDate, ExtractHistory eh, boolean replaceData, Map<String, String> objectCodeMap, Set<String> extractCodes, Map<String, String> pmtTypeCodes) throws Exception {
+		// find all the payments for the given time frame for each vendor
+		PaymentDetailSearch pds = new PaymentDetailSearch();
+		pds.setBeginDisbursementDate(taxYearStartDate);
+		pds.setPayeeIdTypeCd(PayeeIdTypeCodes.VENDOR_ID);
+
+		// add 1 day to end date then truncate time and set as a java.sql.Date
+		pds.setEndDisbursementDate(new java.sql.Date(DateUtils.addDays(DateUtils.truncate(taxYearEndDate, Calendar.DAY_OF_MONTH), 1).getTime()));
+
+		pds.setPayeeId(vendor.getVendorNumber());
+
+		List<PaymentDetail> paymentDetails = taxReporting1099Dao.getAllPaymentsForSearchCriteria(pds);
+
+		if ((paymentDetails != null) && !paymentDetails.isEmpty()) {
+			if (LOG.isInfoEnabled()) {
+				LOG.info(vendor.getVendorName() + ": #payments=" + paymentDetails.size());
+			}
+
+			// Lookup payee by vendor number and tax year - if we don't find and existing Payee we will create a new one
+			Payee payee = getPayeeByVendorNum(vendor, taxYear);
+
+			// process payment where the DocumentIncomeType tables should be used (PREQ, DV, CM) return any payment in the unprocessedPayments that do not fit the criteria
+			List<PaymentDetail> unprocessedPayments = processDocumentIncomeTypePayements(payee.getId(), paymentDetails, replaceData, eh);
+
+			if (LOG.isDebugEnabled()) {
+				if (!unprocessedPayments.isEmpty()) {
+					LOG.debug("found " + unprocessedPayments.size() + " unprocessed payment details");
+				}
+			}
+
+			// processing any remaining payments in the default manner
+			for (PaymentDetail paymentDetail : unprocessedPayments) {
+				// if we are replacing all payment information or this payment information does not already exist
+				if (replaceData || !hasPayments(payee.getId(), paymentDetail)) {
+					eh.incrementPaymentsExtracted(savePayments(objectCodeMap, extractCodes, pmtTypeCodes, payee, paymentDetail));
+				}
+			}
+		} 
+		else if (LOG.isDebugEnabled()) {
+			LOG.debug(vendor.getVendorName() + " : No Payments Found");
+		}
+	}
+
+	// this will be implemented later
```

Comment:
If this is indeed to be implemented later, please append comment to include the UAF-#### of that it depends upon. This makes finding the location simple later on and gives a point of reference for a curious developer happening across this code section.